scripts/scil_filter_tractogram.py

```diff
@@ -0,0 +1,273 @@
+#!/usr/bin/env python
+#  -*- coding: utf-8 -*-
+
+import argparse
+import logging
+import os
+
+from dipy.io.stateful_tractogram import Space, StatefulTractogram
+from dipy.io.streamline import load_tractogram, save_tractogram
+from dipy.io.utils import is_header_compatible
+import nibabel as nib
+import numpy as np
+
+from scilpy.io.utils import (add_overwrite_arg,
+                             assert_inputs_exist,
+                             assert_outputs_exists,
+                             read_info_from_mb_bdo)
+from scilpy.segment.streamlines import (filter_grid_roi,
+                                        filter_ellipsoid,
+                                        filter_cuboid)
+
+DESCRIPTION = """
+    Updated version of the robust filtering script.
+    Now supports sequential filtering condition and mix filtering object.
+    For example, --atlas_roi ROI_NAME ID MODE CRITERIA
+    - ROI_NAME is the filename of a Nifti
+    - ID is the integer value in the atlas
+    - MODE must be one of these values: 'any', 'either_end', 'both_ends'
+    - CRITERIA must be one of these values: ['include', 'exclude']
+
+    Multiple filtering tuples can be used and options mixed.
+    A logical AND is the only behavior available. All theses filtering
+    conditions will be sequentially applied.
+"""
+
+
+def _buildArgsParser():
+    p = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                description=DESCRIPTION)
+
+    p.add_argument('in_tractogram', metavar='IN_TRACTOGRAM',
+                   help='Path of the input tractogram file (.trk)')
+    p.add_argument('out_tractogram', metavar='OUT_TRACTOGRAM',
+                   help='Path of the output tractogram file (.trk)')
+
+    p.add_argument('--reference',
+                   help='Reference anatomy for Tck/Vtk file support')
+
```

Comment:
Should we use the new add_reference from https://github.com/scilus/scilpy/pull/25